app/services/content_change_deduplicator_service.rb

```diff
@@ -0,0 +1,18 @@
+class ContentChangeDeduplicatorService
+  def self.call(*args)
+    new.call(*args)
+  end
+
+  def call(content_changes)
+    deduplicated = reverse_by_public_updated_at(content_changes)
```

Comment:
I wonder if the separate methods are overkill here, I think this is just as readable:

```ruby
class ContentChangeDeduplicatorService
  def self.call(content_changes)
    content_changes
      .sort_by(&:public_updated_at).reverse
      .uniq(&:content_id)
      .sort_by(&:public_updated_at)
  end
end
```

Reply:
I've changed this a bit but not quite how you suggested. I've done the reverse sort in one operation so I've left that as a separate method as it's unclear what it's doing otherwise. I've also remove the last sort as having spoken to @keithiopia offline they want to proceed with descending order of `public_updated_at` for now.